lib/matplotlib/axes/_axes.py

```diff
@@ -5148,20 +5170,33 @@ def fill_between(self, x, y1, y2=0, where=None, interpolate=False,
             where = True
         else:
             where = np.asarray(where, dtype=bool)
-            if where.size != x.size:
+            if where.size != x.size and step != 'between':
                 cbook.warn_deprecated(
                     "3.2",
                     message="The parameter where must have the same size as x "
                             "in fill_between(). This will become an error in "
                             "future versions of Matplotlib.")
-        where = where & ~functools.reduce(np.logical_or,
-                                          map(np.ma.getmask, [x, y1, y2]))
 
-        x, y1, y2 = np.broadcast_arrays(np.atleast_1d(x), y1, y2)
+        if step == 'between':
+            y1, y2, where = cbook.broadcaster([y1, y2, where],
+                                              def_size=x.size - 1)
```

Comment:
Can these not just use `np.broadcast_to`?

Reply:
yeah, don't know why I got locked into thinking the 3 have to be broadcasted in one call